### ISSUE TITLE
`InstanceBehavior` - How does it work?

### ISSUE THREAD

**Issue by JulianEgbert** [hpi-swa-lab/godot-pronto#176]
In the `InstanceBehavior` we have the functions `create_instance()`, `update_scene()` and `maybe_update_scene()` which are all public (visible). To my undestanding this node is primarily used within the editor and not during runtime. Therefore connections to this node shouldn't be possible?! Maybe I also just don't understand this Behavior because we haven't used it before...

Documentation is present in the README and I see the point of copying stuff with this. I don't see yet, how I can trigger the update_scene method though. Maybe the code needs better comments in order to understand what it does?

**Comment by tom95** [comment]
Indeed, it does not have runtime any significance or API. You would select the behavior, hover its connection menu, and press the special instantiate button at the menu's top.

**Comment by JulianEgbert** [comment]
But how do you call the other functions? There is only one button. Or are they called automatically?

**Comment by tom95** [comment]
Yes, they are all called by various callbacks :)

create_instance when the button in the menu is pressed.
maybe_update when the scene is saved.
update when from maybe_update, if the saved scene is our scene.
